src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx

```diff
@@ -56,31 +56,47 @@ const LoginForm: React.FunctionComponent<PropsType> = ({
   const onSubmit = async (data: LoginFormData) => {
     if (!venue) return;
     try {
-      if (venue.requiresTicketCode) await axios.get(codeCheckUrl(data.code));
-      if (venue.requiresEmailVerification)
-        await axios.get(codeCheckUrl(data.email));
-
-      const auth = await signIn(data);
-
-      if (
-        auth.user &&
-        (venue.requiresTicketCode || venue.requiresEmailVerification)
-      ) {
-        firebase
-          .firestore()
-          .doc(`userprivate/${auth.user.uid}`)
-          .get()
-          .then((doc) => {
-            if (auth.user && doc.exists) {
-              firebase
-                .firestore()
-                .doc(`userprivate/${auth.user.uid}`)
-                .update({
-                  codes_used: [...(doc.data()?.codes_used || []), data.code],
-                  date_of_birth: data.date_of_birth,
-                });
-            }
-          });
+      await signIn(data);
+
+      if (venue.access?.includes(VenueAccessType.Codes)) {
+        const result = await _firebase
+          .functions()
+          .httpsCallable("access-checkAccess")({
+          venueId: venue.id,
+          code: data.code,
+        });
```

Comment:
We seem to use this same chunk of code in a number of places. It would be good to refactor it out into a single function under `api/` that is called, rather than defining it's specifics in all of these locations.